### PR TITLE
RF: cross_entropy, implemented for torch

### PR DESCRIPTION
**Pull request by patrick-wilken**
@albertz asked for this, so already sharing.

The most important case `estimated_type="logits"` seems to work, see tests. But it's not yet complete, other `estimated_type`s don't work yet (sparse would need implementation of `gather`). Also, the case of additional dims other than batch and class label dim is not tested. Will continue tomorrow...

### CODE REVIEW

File: returnn/frontend/loss.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+"""
+Loss functions
+"""
+
```
Comment (albertz) [inline returnn/frontend/loss.py:4]:
```suggestion

from __future__ import annotations
```

File: tests/test_torch_frontend.py
Comment (albertz) [inline tests/test_torch_frontend.py:1]: Please stop using this file in the future, and instead make backend independent test. All the tests from here should be rewritten and moved. To `test_rf_base.py`, or maybe we should make separate files for different groups of functionality.

File: returnn/torch/frontend/_backend.py [hunk 2]
```diff
@@ -153,6 +153,59 @@ def log_softmax(tensor: Tensor, *, axis: Dim) -> Tensor:
         out.raw_tensor = torch.log_softmax(tensor.raw_tensor, dim=tensor.dims.index(axis))
         return out
 
+    @staticmethod
+    def softmax_cross_entropy_with_logits(logits: Tensor, targets: Tensor, axis: Dim):
```
Comment (albertz) [inline returnn/torch/frontend/_backend.py:157]:
```suggestion
    def softmax_cross_entropy_with_logits(*, logits: Tensor, targets: Tensor, axis: Dim):
```

File: returnn/frontend/_backend.py [hunk 3]
```diff
@@ -275,6 +275,18 @@ def log_softmax(tensor: Tensor, *, axis: Dim) -> Tensor:
         """
         raise NotImplementedError
 
+    @staticmethod
+    def softmax_cross_entropy_with_logits(logits: Tensor, targets: Tensor, axis: Dim):
```
Comment (albertz) [inline returnn/frontend/_backend.py:279]:
```suggestion
    def softmax_cross_entropy_with_logits(*, logits: Tensor, targets: Tensor, axis: Dim):
```
